src/test/java/com/example/moviemanager/controller/MovieControllerTest.java

```diff
@@ -37,129 +49,215 @@ public class MovieControllerTest {
+        Movie expectedMovie =
+                new Movie(movieId, "Home Alone", "Christmas movie", 1990, 8.5);
+        when(movieService.update(eq(1L), argThat(getMovieArgumentMatcher())))
+                .thenReturn(expectedMovie);
+        String expectedResponseBody = """
+                   {
+                       "id":1,
+                       "title":"Home Alone",
+                       "description":"Christmas movie",
+                       "releaseYear":1990,
+                       "rating":8.5
+                   }
+                """;
 
-        // then
-        mockMvc.perform(put("/movies/{movieId}", movieId)
+        // when
+        String actualResponseBody = mockMvc.perform(put(MOVIE_BY_ID_URL, movieId)
                         .contentType(MediaType.APPLICATION_JSON)
-                        .content(jsonMovie))
+                        .content(request))
                 .andExpect(status().isOk())
-                .andExpect(content().string(expectedJsonMovie));
+                .andReturn().getResponse().getContentAsString();
+
+        // then
+        JSONAssert.assertEquals(expectedResponseBody, actualResponseBody, true);
     }
 
     @Test
     void returns_404_response_when_trying_to_update_non_existing_movie() throws Exception {
         // given
-        int movieId = 1;
-        Movie movie = new Movie(5, "Home Alone", "Christmas movie", 1990, 8.5);
-        String jsonMovie = objectMapper.writeValueAsString(movie);
+        long movieId = 1;
+        String requestBody = objectMapper.writeValueAsString(
+                new UpdateMovieRequest("Home Alone", "Christmas movie", 1990, 8.5));
         String message = "Update failed. Could not find movie by id - " + movieId;
-        doThrow(new MovieNotFoundException(message)).when(movieService).update(movieId, movie);
+        doThrow(new MovieNotFoundException(message)).when(movieService).update(any(Long.class), any(Movie.class));
 
         // then
-        mockMvc.perform(put("/movies/{movieId}", movieId)
+        mockMvc.perform(put(MOVIE_BY_ID_URL, movieId)
                         .contentType(MediaType.APPLICATION_JSON)
-                        .content(jsonMovie))
+                        .content(requestBody))
                 .andExpect(status().isNotFound())
                 .andExpect(content().string(containsString(message)));
     }
 
     @Test
-    void returns_500_response_when_trying_to_use_character_instead_of_int_in_url() throws Exception {
+    void returns_500_response_when_trying_to_delete_movie_without_specifying_its_id() throws Exception {
         // given
         String message = "Request method 'DELETE' is not supported";
 
         // then
-        mockMvc.perform(delete("/movies"))
+        mockMvc.perform(delete(MOVIES_URL))
                 .andExpect(status().isInternalServerError())
                 .andExpect(content().string(containsString(message)));
     }
+
+    private ArgumentMatcher<Movie> getMovieArgumentMatcher() {
```

Comment:
I think idea of this matcher is that the Movie entity fields should match the DTO fields that was passed to controller, and now you check whether movie entity fields match hardcoded single movie values. I think it should be improved and also method naming express the intent after refactoring.

Reply:
Fixed.